src/lib_ccx/ccx_decoders_708.c

```diff
@@ -1046,6 +1046,18 @@ void dtvcc_handle_DFx_DefineWindow(dtvcc_service_decoder *decoder, int window_id
 	if (anchor_horizontal > CCX_DTVCC_SCREENGRID_COLUMNS - col_count)
 		anchor_horizontal = CCX_DTVCC_SCREENGRID_COLUMNS - col_count;
 
+	if (window->is_defined)
+	{
+		if (row_count < window->row_count)
+		{
+			// Remove the oldest row if the row count is reduced
+			for (int i = row_count; i < window->row_count; i++)
+			{
+				dtvcc_window_rollup(decoder, window);
+			}
+		}
+	}
+
```

Comment:
Would it be possible for you to change it to single if instead like:
if (window->is_defined && row_count <...) {
    for (...
}